client/my-sites/stats/stats-shares/index.jsx

```diff
@@ -0,0 +1,121 @@
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import { useSelector } from 'react-redux';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { Card } from '@automattic/components';
+import StatsTabs from '../stats-tabs';
+import StatsTab from '../stats-tabs/tab';
+import SectionHeader from 'calypso/components/section-header';
+import QuerySiteStats from 'calypso/components/data/query-site-stats';
+import {
+	isRequestingSiteStatsForQuery,
+	getSiteStatsNormalizedData,
+	hasSiteStatsQueryFailed,
+} from 'calypso/state/stats/lists/selectors';
+import { useTranslate } from 'i18n-calypso';
+
+const StatShares = ( { siteId } ) => {
+	const translate = useTranslate();
+	const isLoading = useSelector( ( state ) =>
+		isRequestingSiteStatsForQuery( state, siteId, 'stats' )
+	);
+	const hasError = useSelector( ( state ) => hasSiteStatsQueryFailed( state, siteId, 'stats' ) );
+	const {
+		shares,
+		sharesFacebook,
+		sharesPressThis,
+		sharesTwitter,
+		sharesTumblr,
+		sharesLinkedin,
+		sharesEmail,
+	} = useSelector( ( state ) => getSiteStatsNormalizedData( state, siteId, 'stats' ) ) || {};
```

Comment:
What about other sharing services available in `Sharing_Service::get_all_services` (`wp-content/mu-plugins/post-flair/sharing/sharing-service.php`) like `sharePrint` or `shareTelegram`?

Reply:
Oh, it also seems there is a `include_custom` option... Again, I think there are two options here:

- hardcode in Calypso all known services.
- try creating a catch-all component which will loop over all the `shares_provider` keys and will return `Provider` as the name. 

Whats your pick? Or any other ideas?

Reply:
I think we can somehow implement a dynamic logic.

- According to the stats endpoint (`public.api/rest/wpcom-json-endpoints/class.wpcom-json-api-stats-endpoint.php#L178`), the `shares_*` keys follow a `shares_<SERVICE_ID>` format, so can infer the Service ID from the API response.
- Calypso is able to map this ID to the actual provider name. For instance the `sharing-buttons` endpoints returns a map that associates the service ID with a service name: 
<img width="809" alt="Screen Shot 2021-05-19 at 17 37 36" src="https://user-images.githubusercontent.com/1233880/118841811-ec28d100-b8c8-11eb-8fe9-9b45ec190da5.png">. Could we maybe reuse this API response? Alternatively, we could look into implementing a new API endpoint.

Reply:
Thanks for pointing out his endpoint! It is super helpful! I am now adding the providers dynamically. f18e545